include/mbgl/util/immutable.hpp

```diff
@@ -108,11 +108,18 @@ class Immutable {
 
     template <class S> friend class Immutable;
     template <class S, class U> friend Immutable<S> staticImmutableCast(const Immutable<U>&);
+    template <class S, class U> friend Immutable<S> dynamicImmutableCast(const Immutable<U>&);
 };
 
 template <class S, class U>
 Immutable<S> staticImmutableCast(const Immutable<U>& u) {
     return Immutable<S>(std::static_pointer_cast<const S>(u.ptr));
 }
 
+template <class S, class U>
+Immutable<S> dynamicImmutableCast(const Immutable<U>& u) {
+    (void)dynamic_cast<const S&>(*u); // Call for side effect of failure.
```

Comment:
When the `dynamic_cast` fails, it'll throw a `std::bad_cast` exception, since the type is a reference type (rather than a pointer type). Can you please clarify this in the comment?